examples/darwin-framework-tool/commands/clusters/ReportCommandBridge.h

```diff
@@ -241,3 +241,68 @@ class SubscribeEvent : public ModelCommand {
     uint16_t mMinInterval;
     uint16_t mMaxInterval;
 };
+
+class ReadEvent : public ModelCommand {
+public:
+          ReadEvent()
+        : ModelCommand("read-event-by-id")
+    {
+        AddArgument("cluster-id", 0, UINT32_MAX, &mClusterId);
+        AddArgument("event-id", 0, UINT32_MAX, &mEventId);
+        AddArgument("fabric-filtered", 0, 1, &mFabricFiltered);
+        ModelCommand::AddArguments();
+    }
+
+    ReadEvent(chip::ClusterId clusterId)
+        : ModelCommand("read-event-by-id")
+        , mClusterId(clusterId)
+    {
+        AddArgument("event-id", 0, UINT32_MAX, &mEventId);
+        AddArgument("fabric-filtered", 0, 1, &mFabricFiltered);
+        ModelCommand::AddArguments();
+    }
+
+    ReadEvent(const char * _Nonnull eventName)
+        : ModelCommand("read-event")
+    {
+        AddArgument("event-name", eventName);
+        AddArgument("fabric-filtered", 0, 1, &mFabricFiltered);
+        ModelCommand::AddArguments();
+    }
+
+    ~ReadEvent() {}
+
+    CHIP_ERROR SendCommand(MTRBaseDevice * _Nonnull device, chip::EndpointId endpointId) override
+    {
+        dispatch_queue_t callbackQueue = dispatch_queue_create("com.chip.command", DISPATCH_QUEUE_SERIAL);
+        MTRReadParams * params = [[MTRReadParams alloc] init];
+        if (mFabricFiltered.HasValue()) {
+            params.filterByFabric = mFabricFiltered.Value();
+        }
+        [device
+            readEventsWithEndpointID:[NSNumber numberWithUnsignedShort:endpointId]
+                               clusterID:[NSNumber numberWithUnsignedInteger:mClusterId]
+                             eventID:[NSNumber numberWithUnsignedInteger:mEventId]
+                                  params:params
+                                   queue:callbackQueue
+                              completion:^(NSArray<NSDictionary<NSString *, id> *> * _Nullable values, NSError * _Nullable error) {
+                                  if (error != nil) {
+                                      LogNSError("Error reading event", error);
+                                  }
+                                  if (values) {
+                                      for (id item in values) {
+                                          NSLog(@"Response Item: %@", [item description]);
```

Comment:
This doesn't seem "logging-compatible" with the "read-event" bits in chip-tool.  Is it expected to be?  Will certification expect it to be?

Reply:
I am just following the existing darwin-framework-tool logging style for it. Because we will use our iOS controller app log in certification, the darwin-framework-tool log is not important for certification itself. I just implement it as example to use the new matter framework API "readEventsWithEndpointID". If some modification is needed, please let me know.

Reply:
Please check with @krypton36 what he wants for darwin-framework-tool here?

Reply:
@krypton36 could you give some advice of this code change? I want to show how to use the new API readEventsWithEndpointID to the other app developers and to verify my implementation of it by using darwin-framework-tool.